ra_aid/agents/ciayn_agent.py

```
@@ -248,5 +342,42 @@ def stream(self, messages_dict: Dict[str, List[Any]], config: Dict[str, Any] = N
+
+```
+{code}
+```
+    """
+    response = model.invoke(prompt)
+    response = response.content
+
+    pattern = r"([\w_\-]+)\((.*?)\)"
+    matches = re.findall(pattern, response, re.DOTALL)
+    if len(matches) == 0:
+        raise ToolExecutionError("Failed to extract tool call")
```

Ruff (0.8.2)

Avoid specifying long messages outside the exception class

(TRY003)